example/src/eyedrop/getCanvasBlockColors.js

```diff
@@ -0,0 +1,28 @@
+// returns an array of rgb objects [{r:10, g:10, b:10}]
+const getCanvasBlockColors = (canvas, sx, sy, sw, sh) => {
+    if (!canvas.getContext) {
+        throw new Error("Input element is not a canvas")
+    }
+
+    const width = canvas.width;
+    const height = canvas.height;
+
+    const ctx = canvas.getContext('2d');
+
+    // do not compute data if they're out of canvas area
+    if (sx < 0) { sx = 0 }
+    if (sy < 0) { sy = 0 }
+    if (sx + sw > width) { sw = width - sx }
+    if (sy + sh > height) { sh = height - sy }
```

Comment:
The names! :smile:

Reply:
(sx, sy, sw, sh)

Reply:
This function is generic, it is not specific for the eyedropper pick. So I think leaving the names as generic here is better.

Reply:
Hmm... Alright. But let's still write the names out - generic or not, we can _always_ be verbose. 😉